### PR TITLE
Limits adventurer slots, lowers mercenary slots

### PR DESCRIPTION
**Pull request by oprimeolt**
## About The Pull Request
Swaps adventurer roles from infinite to 8, wanted to do 6 like old RT but this way there can be parties. Pilgrim is still an infinite slot midround respawn role. Limited merc slots from 5 each to 2 each. 

## Why It's Good For The Game
Limits slots that are 'town role but better' or 'guard role but better' so the actual town and guard roles are played

### CODE REVIEW

**Comment by EaglePhntm** [comment]
limit adventurer and kill the pop.

**Comment by Mannybrado** [comment]
it should be noted that the late-spawn drifter system still allows for infinite adventurers. you just have to wait to get a party that consists of them if you reeeeeally want it.

**Comment by oprimeolt** [comment]
> it should be noted that the late-spawn drifter system still allows for infinite adventurers. you just have to wait to get a party that consists of them if you reeeeeally want it.

yeah this

**Comment by EaglePhntm** [comment]
> > it should be noted that the late-spawn drifter system still allows for infinite adventurers. you just have to wait to get a party that consists of them if you reeeeeally want it.
> 
> yeah this

i dont see the point of this then really.

**Comment by Mannybrado** [comment]
> > > it should be noted that the late-spawn drifter system still allows for infinite adventurers. you just have to wait to get a party that consists of them if you reeeeeally want it.
> > 
> > 
> > yeah this
> 
> i dont see the point of this then really.

still a limiting factor that should exist tbh. infinite combat roles at all times with no caveats is dumb for what this server is trying to be

**Comment by Mannybrado** [comment]
the drifter system is significantly more likely to spawn pilgrims than adventurers but you can see what it'll spawn

if it means more people will try out heartfelt(should remove the race limits) or bandits or w/e, then good
also upstream is adding more unique drifter parties so that should be good

**Comment by CVSthePharmacy** [comment]
OPRIME WHAT ARE YOU DOING THIS ISN'T F13 THE TOWN IS NOT YOUR BOS

**Comment by CVSthePharmacy** [comment]
> it should be noted that the late-spawn drifter system still allows for infinite adventurers. you just have to wait to get a party that consists of them if you reeeeeally want it.

also it's broken, doesn't let me join as a heartfelt knight when I select it....

**Comment by EaglePhntm** [comment]
"cancer" that keeps the game alive.

**Comment by Xehra52** [comment]
Perhaps limiting it is not the way to go... Disincentivising playing to be just towner+ would be better, perhaps make adventurer's more combat focused and making them unable to perform the tasks/roles of towner whilst keeping their stats would be better... For example, to be constructive here, making them capped at 'apprentice' or similar in skills that do not befit an 'adventurer' of their class... For example, a barbarian should not be able to create a legendary masterpiece of metal when he can't understand how something dies beyond simply hitting it hard enough.

**Comment by EaglePhntm** [comment]
> Perhaps limiting it is not the way to go... Disincentivising playing to be just towner+ would be better, perhaps make adventurer's

just make other roles more appealing in skills than making adventurer ass honestly. Barbarian example, that -2 intelligence may already cause most to not even be able to make a wood pipe let alone learn any skills before round ends

**Comment by oprimeolt** [comment]
> OPRIME WHAT ARE YOU DOING THIS ISN'T F13 THE TOWN IS NOT YOUR BOS

no other RT server I wanna play rt

**Comment by Mannybrado** [comment]
> also it's broken, doesn't let me join as a heartfelt knight when I select it....

heartfelt roles are human only, someone sould probably change that

**Comment by CVSthePharmacy** [comment]
> > also it's broken, doesn't let me join as a heartfelt knight when I select it....
> 
> heartfelt roles are human only, someone sould probably change that

lore accurate

**Comment by coffeecoffeecoffee** [comment]
> Also this conversation has been talked to DEATH. Adventurers have been a problem from RT start.
> 
> Adventurers, however, are a cancer. No-one cares about pilgrims, even if they're a no-requirement and decent stats and skills role.
> 
> There's been a replication of lifeweb through out RT's history. It will never really achieve it, in spite of it's attempts to and I will forever state until this code-base dies, it is an unfinished piece of shit with lepers and random shit pushing around in it.
> 
> And the main reasons no-one plays town is purely because the town sucks. There's nothing to do other than be a retard bumbling around in the streets. I genuinely do not believe, even if the design of the town was overhauled, that people would actually want to play a non-combat role. There is an inherent issue with the design and that's assuming people will willingly play non-combat roles.
> 
> Besides, what's there to do? Farm for 5 hours? Mine for 5 hours? Smith for 5 hours? Get murdered five minutes in? There is nothing to actually do. It's labor to keep you busy until you finally decide, to go fuck and bar-RP at the Inn.

why even comment on a codebase you're so sure is an "unfinished piece of shit" and will stay that way "until this code-base dies"

adventurers are not a problem, they're a boogeyman, just like assistants, just like every time people have tried to limit assistants, in servers past. if adventurers have _balancing problems,_ that's a different and very valid thing.

**Comment by Ruediger4** [comment]
> 
> This is a terrible reply.
> 
> Here's why, because previously when RT allowed adventurers to be unlimited, they would run around being annoying little shits and the town was always empty. Adventurer needs to be limited because it isn't like an assistant, they've got actual meaningful impact on the game by being completely immune to the king's master rod so if you want to kill the king at any point, just be an adventurer along with having the combat skills to stack up decently against anyone with a okay-ish garrison role.
> 
> And they have zero responsibility. This wouldn't be bad if there wasn't so many running around being mini-combatants but no-one is asking for Pilgrim to be limited, and they've still got decent stats and skills.
> 
> All of this would be excusable if they weren't being used to act like NPCs with zero reason to exist other than to build a shack because they didn't want to pay 5 coins from their pouch.
> 
> In fact this problem might worse if hedge knights appear.
> 
> The further compounding of all of these issues is, well, the fact that town is in dire need of content to encourage to people to play it. It is unfinished and the volunteers (Us) are left to finish it because zeth has fucked off to RT2. That was my point, not whatever point you managed to string from mis-reading it.

this feels like the "Heartbreaking: the worst person you know is right", but because I wish it wasn't true. IDK half the maintainers here but @Jeff-Dome do be right that Adventurers should be a bit more impactful/"unique" (Paladins being Templars but w/o direct Church allegiance, Clerics being something like combat acolytes, Barbarian/Fighter being a stand-in for Every Combat Role Ever, etc.) but everyone and their mother goes as one and just, kinda don't do a lot?

mostly because there isn't anyone/anything that will actually try to push them back if they behave like shits since rarely are there a 'high' number of Watchmen/Bog guards. And when there IS, the garrison is usually 10x more powerful than them so adventurers just decide to lay low and not do anything but stack coin and skills and... do nothing. Usually.

And Adventurers learning skills because there aren't any Carpenters/Smiths/Seamstress isn't inherently wrong. The issue is those roles tend to just fill the "be a bit of a NPC" gap unless you know you'll get a lot of people coming for you (Smith). I do agree that maybe adventurers should require more XP to level up their skills or be restricted to apprentice at most. It'd make Townies feel more significant/impactful.

Hell, in a way, **players fuck themselves from getting any RP by avoiding finding a townie whose life is carpentery/sewing/weaving/smithing just to get 'things' done 'faster'.** The amount of times someone has gone "I wish building a house didnt take forever ://" when you could've just gone and found a carpenter and done meaningful RP **AND** give a player something to do is BAFFLING.

So yeah, maybe limiting adventurer numbers to 10??? and then leave Migration for RNG luck.

Don't shaft merc numbers tho.
I'm biased.
I'm really biased.
I like mercs.

*tldr lower adventurer numbers good, lower merc numbers bad. I cry.*

**Comment by Wesreidau** [comment]
Players wanting to roleplay a consistent character will want their slot to be available to them. We have no way to sort the fraggers from the roleplayers beyond PQ requirements and banning. I would rather see a round of fifty amazonians and barbarians all roleplaying a consistent and serious character tied to the class they picked than a mishmash of Grognack the Carpenter and Serena Bloodaxe the Cheesemaker.

Let people play the thing they want to play and tweak the game to match. For example; add more things for adventurers to do, like dungeons and looting, that feed back into things the town and castle need, like vault valuables and plunder. Weaken adventurer starting skills but develop training from combat with mobs, decreasing their effectiveness against soldiers-with-jobs unless and until they fight their way up. Make dungeon monsters, that is those who won't spawn from ambushes and kill non-combats, more lethal to start meat-grinding adventurers.

Our ideal adventurer experience is a band of idiots who go into a dungeon as a regular DnD party and come back out with a pile of loot to sell to the smiths, merchants and stewards. Ideally they'll be missing a few members every return trip, until only Grognak is left alive with his master swordsmanship and crippling PTSD in the inn wasting his loot on nightswains.

People want freedom and to be their snowflakes, so make their life harder in that freedom and melt some snowflakes. Make adventurer grindy. Then since the incentive changed, they'll get a real job for the character.

**Comment by coffeecoffeecoffee** [comment]
> Players wanting to roleplay a consistent character will want their slot to be available to them. We have no way to sort the fraggers from the roleplayers beyond PQ requirements and banning. I would rather see a round of fifty amazonians and barbarians all roleplaying a consistent and serious character tied to the class they picked than a mishmash of Grognack the Carpenter and Serena Bloodaxe the Cheesemaker.
> 
> Let people play the thing they want to play and tweak the game to match. For example; add more things for adventurers to do, like dungeons and looting, that feed back into things the town and castle need, like vault valuables and plunder. Weaken adventurer starting skills but develop training from combat with mobs, decreasing their effectiveness against soldiers-with-jobs unless and until they fight their way up. Make dungeon monsters, that is those who won't spawn from ambushes and kill non-combats, more lethal to start meat-grinding adventurers.
> 
> Our ideal adventurer experience is a band of idiots who go into a dungeon as a regular DnD party and come back out with a pile of loot to sell to the smiths, merchants and stewards. Ideally they'll be missing a few members every return trip, until only Grognak is left alive with his master swordsmanship and crippling PTSD in the inn wasting his loot on nightswains.
> 
> People want freedom and to be their snowflakes, so make their life harder in that freedom and melt some snowflakes. Make adventurer grindy. Then since the incentive changed, they'll get a real job for the character.

The reply we needed, holy shit. Someone with some sense and imagination.

**Comment by DerFlammenwerfer** [comment]
> It's main character syndrome manifest. Everyone wants to have an epic fragger role and pretend it's not purely about wanting epic frag skills but the RP and the Character they've made.
> 
> You see billions of warriors, amazonians and clerics but never one pilgrim. Funny that.

The same people playing 'epic fragger roles' are also usually the ones doing most of the roleplaying because, go figure, they make up the majority of the population, **which is nothing new in this downstream**. Hell, I play a nature cleric, but this same character started out as a Dendor-worshipping hunter, and it just so happens that the spells Nature clerics get actually benefit my playstyle and fit the theme of the character I'm trying to play -- a supporting role.

As for why you never see pilgrims? It's mostly because they are literally just towners who spawn in the bog.

**Comment by kyowaaaa** [comment]
> Players wanting to roleplay a consistent character will want their slot to be available to them. We have no way to sort the fraggers from the roleplayers beyond PQ requirements and banning. I would rather see a round of fifty amazonians and barbarians all roleplaying a consistent and serious character tied to the class they picked than a mishmash of Grognack the Carpenter and Serena Bloodaxe the Cheesemaker.
> 
> Let people play the thing they want to play and tweak the game to match. For example; add more things for adventurers to do, like dungeons and looting, that feed back into things the town and castle need, like vault valuables and plunder. Weaken adventurer starting skills but develop training from combat with mobs, decreasing their effectiveness against soldiers-with-jobs unless and until they fight their way up. Make dungeon monsters, that is those who won't spawn from ambushes and kill non-combats, more lethal to start meat-grinding adventurers.
> 
> Our ideal adventurer experience is a band of idiots who go into a dungeon as a regular DnD party and come back out with a pile of loot to sell to the smiths, merchants and stewards. Ideally they'll be missing a few members every return trip, until only Grognak is left alive with his master swordsmanship and crippling PTSD in the inn wasting his loot on nightswains.
> 
> People want freedom and to be their snowflakes, so make their life harder in that freedom and melt some snowflakes. Make adventurer grindy. Then since the incentive changed, they'll get a real job for the character.

Absolutely love this thought, it sums up what I've wanted to say but way better. If people wanna be vagabonds with swords, make it more dangerous and give rewards that help the other roles too.
